cpp/include/cudf/lists/detail/slice.hpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/lists/lists_column_view.hpp>
+
+namespace cudf {
+namespace lists {
+namespace detail {
+std::unique_ptr<cudf::column> slice(lists_column_view const& list,
```

Comment:
Where should tests for this go?

Reply:
tests/lists/slice_tests.cpp ???

Reply:
`cpp/tests/copying`

Reply:
Since this is really an internal function that is only ever used by the `column(column_view)` constructor, I put tests in `column_tests.cu`. I'm not sure about the name `slice` - since `slice` is also used for the API that does zero-copy slicing - but that's the name that the equivalent strings function uses. @davidwendt would you have any advice here?

Reply:
The name of the strings function I think you are referring to in libcudf is actually called [`cudf::strings::slice_strings()`](https://docs.rapids.ai/api/libcudf/stable/group__strings__substring.html#gaa689719250048ea788b5db1549c8265f) so as to not confuse it with `cudf::slice()`

[There is a `cudf::strings::detail::slice()` that is called by `column(column_view)`](https://github.com/rapidsai/cudf/blob/ab1a1e8964f5e5b30f8aa08e81764234f24024cf/cpp/src/column/column.cu#L187) -- in the same way as `cudf::lists::detail::slice()` in [this PR](https://github.com/rapidsai/cudf/blob/ab1a1e8964f5e5b30f8aa08e81764234f24024cf/cpp/src/column/column.cu#L240).

I think adding the tests for lists for `column(column_view)` into `column_test.cu` as you have done makes sense to me.

Reply:
Right -- I guess my worry is that the `detail::slice()` functions could be confused with the public `cudf::slice()` function, and I was wondering whether we should change both `strings::detail::slice()` and `lists::detail::slice()` to something else (`copy_slice()` perhaps?)

Reply:
I think renaming these to `detail::copy_slice()` is a good idea. The `cudf::strings::detail::slice()` was just created before `cudf::slice()` was created/ported.

Reply:
Thoroughly agree.